Model/Payment/BankSlip.php

```diff
@@ -0,0 +1,194 @@
+        \Magento\Framework\Api\ExtensionAttributesFactory $extensionFactory,
+        \Magento\Framework\Api\AttributeValueFactory $customAttributeFactory,
+        \Magento\Payment\Helper\Data $paymentData,
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
+        \Magento\Payment\Model\Method\Logger $logger,
+        \Magento\Framework\Model\ResourceModel\AbstractResource $resource = null,
+        \Magento\Framework\Data\Collection\AbstractDb $resourceCollection = null,
+        array $data = []
+    ) {
+
+        $this->_logger = $logger;
+        parent::__construct(
+            $context,
+            $registry,
+            $extensionFactory,
+            $customAttributeFactory,
+            $paymentData,
+            $scopeConfig,
+            $logger,
+            $resource,
+            $resourceCollection,
+            $data
+        );
+
+        $this->api = $api;
+        $this->_invoiceService = $invoiceService;
+        $this->customer = $customer;
+        $this->product = $product;
+        $this->bill = $bill;
+        $this->paymentMethod = $paymentMethod;
+        $this->date = $date;
+        $this->psrLogger = $psrLogger;
+        $this->profile = $profile;
+    }
+
+
+    public function isAvailable(
+        \Magento\Quote\Api\Data\CartInterface $quote = null
+    ) {
+
+        return parent::isAvailable($quote);
+    }
+
+    /**
+     * Assign data to info model instance
+     *
+     * @param mixed $data
+     *
+     * @return BankSlip
+     */
+    public function assignData(DataObject $data)
+    {
+        $info = $this->getInfoInstance();
+        $info->setAdditionalInformation('installments', 1);
+        $info->save();
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getPaymentMethodCode()
+    {
+        return PaymentMethod::BANK_SLIP;
+    }
+
+    public function authorize(\Magento\Payment\Model\InfoInterface $payment, $amount)
+    {
+        /** @var Order $order */
+        $order = $payment->getOrder();
+        $customerId = $this->customer->findOrCreate($order);
+        $productList = $this->product->findOrCreateProducts($order);
+
+        $body = [
+            'customer_id' => $customerId,
+            'payment_method_code' => $this->getPaymentMethodCode(),
+            'bill_items' => $productList
+        ];
+
+        if ($installments = $payment->getAdditionalInformation('installments')) {
+            $body['installments'] = (int)$installments;
+        }
+
+        if ($bill = $this->bill->create($body)) {
+            if ($bill['charges'][0]['payment_method']['code'] === PaymentMethod::BANK_SLIP) {
+                $payment->setAdditionalInformation('print_url', $bill['charges'][0]['print_url']);
+                $payment->setAdditionalInformation('due_at', $bill['charges'][0]['due_at']);
+            }
+
+            if ($bill['charges'][0]['payment_method']['code'] === PaymentMethod::BANK_SLIP
```

Comment:
Faz sentido abstrair os comportamentos semelhantes em outro local? pois a mesma validação é realizada no cartão de crédito.

Reply:
As validações diferem um pouco, e notei também enquanto testava que ao criar uma Bill (Fatura), no retorno da API o parâmetro `code` vem nulo, que deveria vir o método de pagamento usado, acredito que funcionava antes pois o @leonardola faz a validação usando ele quando desenvolveu o método de pagamento de cartão de crédito. A alternativa que encontrei foi usar o retorno `$bill['charges'][0]['payment_method']['code']`.

Reply:
Nesse caso, não faz sentido realizar essa validação aqui, pois já estamos dentro do **Model/Payment/BankSlip.php** e obrigatóriamente o método de pagamento dessa fatura deve ser Boleto.

Essa validação faz sentido se for realizada uma abstração como por [exemplo](https://github.com/vindi/vindi-magento/blob/master/app/code/community/Vindi/Subscription/Trait/PaymentProcessor.php)

Reply:
:tada: perfeito :balloon: